libdrgn/dwarf_index.c

```diff
@@ -1426,6 +1444,10 @@ index_cu_second_pass(struct drgn_dwarf_index_namespace *ns,
 		    !specification) {
 			if (insn & DIE_FLAG_DECLARATION)
 				declaration = true;
+			if (signature)
+				// TODO determine if this is needed - usually types with signature are also declarations
```

Comment:
From what I can tell, a DIE with `DW_AT_signature` but not `DW_AT_declaration` is still a definition, which is pointless at best and nonsense at worst. So, I don't think we need this.